### PR TITLE
raven

### PR DESCRIPTION
**Pull request by erinyoung**
raven is a cool de novo assembler for ONT sequencing

### CODE REVIEW

**Comment by kapsakcj** [comment]
Thanks for the PR!

One suggestion, is to update the base image to `ubuntu:bionic` which has `cmake` v3.10.2 available via `apt-get`: https://packages.ubuntu.com/bionic/cmake

Could likely reduce the size of the docker image and simplify the install since you won't have to install `cmake` manually.

Totally up to you.

File: raven/1.5.1/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# install dependencies via apt-get or yum if using a centos or fedora base
+RUN apt-get update && apt-get install -y \
+ wget \
+ git \
+ clang \
+ zlib1g-dev \
+ gcc \
+ build-essential \
+ libssl-dev \
+ libcurl4-openssl-dev
```
Comment (kapsakcj) [inline raven/1.5.1/Dockerfile:28]: could you please add this to the end of the apt-get install layer ?` && apt-get autoclean && rm -rf /var/lib/apt/lists/* `

**Comment by erinyoung** [comment]
It'll probably build A LOT faster with `ubuntu:bionic`. Thank you for the tip!

I'll try that and get back with you.

**Comment by erinyoung** [comment]
Actually, raven appears to need cmake > 3.11. 

```
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.11 or higher is required.  You are running version 3.10.2
```

I think I'll keep the manual install of cmake.

**Comment by kapsakcj** [comment]
upgrade again to ubuntu:focal 😅 https://packages.ubuntu.com/focal/cmake

should be cmake v3.16.3

File: raven/1.5.1/Dockerfile (hunk 1 above)
Comment (erinyoung) [reply raven/1.5.1/Dockerfile]: done!

**Comment by erinyoung** [comment]
I need to learn your secret to how you are finding these...

**Comment by kapsakcj** [comment]
packages.ubuntu.com is your best friend here. This site lists all packages that are available via apt/apt-get

Searching functionality isn't the best but you can search by Ubuntu distribution, so Xenial = Ubuntu 16, bionic = Ubuntu 18, focal = Ubuntu 20.

Later Ubuntu distros will have more up-to-date packages & versions.

You have to be pretty exact with the package names you are searching for. You can find perl libraries here too, the package names often end in `-perl`. This will avoid having to install perl libraries with `cpan` or `cpanm`

**Comment by erinyoung** [comment]
Ta da! Now it's ready. It's definitely faster to build when not building cmake from source.

This container appears to work "for me".

**Comment by kapsakcj** [comment]
Seeing as this is a new container, I don't think we need to do extensive testing. I'll go ahead and merge and build on dockerhub

**Comment by kapsakcj** [comment]
raven comes with a quick test, which passed fine for me in the container:
```
root@3011bfa3eb4a:/data# /raven-1.5.1/build/bin/raven_test
Running main() from /raven-1.5.1/build/_deps/googletest-src/googletest/src/gtest_main.cc
[==========] Running 2 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 2 tests from RavenTest
[ RUN      ] RavenTest.Assemble
[raven::Graph::Construct] minimized 0 - 236 / 236 0.132463s
[raven::Graph::Construct] mapped sequences 0.134580s
[raven::Graph::Construct] annotated piles 0.003651s
[raven::Graph::Construct] removed contained sequences 0.000369s
[raven::Graph::Construct] removed chimeric sequences 0.001149s
[raven::Graph::Construct] minimized 0 - 35 / 35 0.110015s
[raven::Graph::Construct] mapped valid sequences 0.062153s
[raven::Graph::Construct] updated overlaps 0.000021s
[raven::Graph::Construct] removed false overlaps 0.000844s
[raven::Graph::Construct] stored 60 nodes 0.001966s
[raven::Graph::Construct] stored 316 edges 0.000167s
[raven::Graph::Construct] 0.447380s
[raven::Graph::Assemble] removed transitive edges 0.000323s
[raven::Graph::Assemble] removed tips and bubbles 0.000017s
[raven::Graph::Assemble] removed long edges 0.000179s
[raven::Graph::Assemble] 0.001285s
[racon::Polisher::Polish] minimized 0 - 1 / 1 0.050558s
[racon::Polisher::Polish] mapped sequences 0.123835s
[racon::Polisher::Polish] found 181 overlaps 0.179658s
[racon::Polisher::Polish] reverse complemented sequences 0.000030s
[racon::Polisher::Polish] aligned 181 / 181 overlaps [================] 1.050697s
[racon::Polisher::Polish] prepared 93 window placeholders 0.002622s
[racon::Polisher::Polish] called consensus for 93 / 93 windows [================] 1.237961s
[racon::Polisher::Polish] 2.471939s
[racon::Polisher::Polish] minimized 0 - 1 / 1 0.046135s
[racon::Polisher::Polish] mapped sequences 0.133367s
[racon::Polisher::Polish] found 194 overlaps 0.183119s
[racon::Polisher::Polish] reverse complemented sequences 0.000006s
[racon::Polisher::Polish] aligned 194 / 194 overlaps [================] 0.956247s
[racon::Polisher::Polish] prepared 96 window placeholders 0.002564s
[racon::Polisher::Polish] called consensus for 96 / 96 windows [================] 1.216518s
[racon::Polisher::Polish] 2.359419s
[       OK ] RavenTest.Assemble (5366 ms)
[ RUN      ] RavenTest.Checkpoints
[raven::Graph::Construct] minimized 0 - 236 / 236 0.125858s
[raven::Graph::Construct] mapped sequences 0.120840s
[raven::Graph::Construct] annotated piles 0.003029s
[raven::Graph::Construct] removed contained sequences 0.000268s
[raven::Graph::Construct] removed chimeric sequences 0.000652s
[raven::Graph::Construct] minimized 0 - 35 / 35 0.085921s
[raven::Graph::Construct] mapped valid sequences 0.048443s
[raven::Graph::Construct] updated overlaps 0.000029s
[raven::Graph::Construct] removed false overlaps 0.000812s
[raven::Graph::Construct] stored 60 nodes 0.002064s
[raven::Graph::Construct] stored 316 edges 0.000254s
[raven::Graph::Construct] 0.388171s
[raven::Graph::Assemble] removed transitive edges 0.000388s
[raven::Graph::Assemble] removed tips and bubbles 0.000017s
[raven::Graph::Assemble] removed long edges 0.000194s
[raven::Graph::Assemble] 0.001592s
[racon::Polisher::Polish] minimized 0 - 1 / 1 0.075864s
[racon::Polisher::Polish] mapped sequences 0.116331s
[racon::Polisher::Polish] found 181 overlaps 0.196034s
[racon::Polisher::Polish] reverse complemented sequences 0.000014s
[racon::Polisher::Polish] aligned 181 / 181 overlaps [================] 1.050474s
[racon::Polisher::Polish] prepared 93 window placeholders 0.002666s
[racon::Polisher::Polish] called consensus for 93 / 93 windows [================] 1.209276s
[racon::Polisher::Polish] 2.459181s
[racon::Polisher::Polish] minimized 0 - 1 / 1 0.055271s
[racon::Polisher::Polish] mapped sequences 0.131476s
[racon::Polisher::Polish] found 194 overlaps 0.190462s
[racon::Polisher::Polish] reverse complemented sequences 0.000007s
[racon::Polisher::Polish] aligned 194 / 194 overlaps [================] 0.951021s
[racon::Polisher::Polish] prepared 96 window placeholders 0.002905s
[racon::Polisher::Polish] called consensus for 96 / 96 windows [================] 1.189261s
[racon::Polisher::Polish] 2.334128s
[raven::Graph::Construct] minimized 0 - 236 / 236 0.128469s
[raven::Graph::Construct] mapped sequences 0.119689s
[raven::Graph::Construct] annotated piles 0.003321s
[raven::Graph::Construct] removed contained sequences 0.000315s
[raven::Graph::Construct] removed chimeric sequences 0.000819s
[raven::Graph::Construct] reached checkpoint 0.001349s
[raven::Graph::Construct] minimized 0 - 35 / 35 0.087398s
[raven::Graph::Construct] mapped valid sequences 0.050821s
[raven::Graph::Construct] updated overlaps 0.000020s
[raven::Graph::Construct] removed false overlaps 0.000782s
[raven::Graph::Construct] stored 60 nodes 0.001834s
[raven::Graph::Construct] stored 316 edges 0.000196s
[raven::Graph::Construct] reached checkpoint 0.002123s
[raven::Graph::Construct] 0.397136s
[raven::Graph::Assemble] removed transitive edges 0.000249s
[raven::Graph::Assemble] reached checkpoint 0.001787s
[raven::Graph::Assemble] removed tips and bubbles 0.000021s
[raven::Graph::Assemble] reached checkpoint 0.004962s
[raven::Graph::Assemble] removed long edges 0.000171s
[raven::Graph::Assemble] reached checkpoint 0.003331s
[raven::Graph::Assemble] 0.011275s
[racon::Polisher::Polish] minimized 0 - 1 / 1 0.051517s
[racon::Polisher::Polish] mapped sequences 0.114503s
[racon::Polisher::Polish] found 181 overlaps 0.169676s
[racon::Polisher::Polish] reverse complemented sequences 0.000019s
[racon::Polisher::Polish] aligned 181 / 181 overlaps [================] 1.057117s
[racon::Polisher::Polish] prepared 93 window placeholders 0.003054s
[racon::Polisher::Polish] called consensus for 93 / 93 windows [================] 1.206645s
[racon::Polisher::Polish] 2.437483s
[raven::Graph::Polish] reached checkpoint 0.000539s
[racon::Polisher::Polish] minimized 0 - 1 / 1 0.050478s
[racon::Polisher::Polish] mapped sequences 0.124315s
[racon::Polisher::Polish] found 194 overlaps 0.178225s
[racon::Polisher::Polish] reverse complemented sequences 0.000005s
[racon::Polisher::Polish] aligned 194 / 194 overlaps [================] 0.950332s
[racon::Polisher::Polish] prepared 96 window placeholders 0.002845s
[racon::Polisher::Polish] called consensus for 96 / 96 windows [================] 1.176690s
[racon::Polisher::Polish] 2.309051s
[raven::Graph::Polish] reached checkpoint 0.001292s
[       OK ] RavenTest.Checkpoints (10484 ms)
[----------] 2 tests from RavenTest (15850 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test suite ran. (15850 ms total)
[  PASSED  ] 2 tests.
```

**Comment by kapsakcj** [comment]
now up on dockerhub and quay

https://hub.docker.com/r/staphb/raven

https://quay.io/repository/staphb/raven?tab=tags

**Pull request merged**